Problem building expo-gl-cpp with unimodules on android

### Summary

Please provide the following:

SDK Version: expo-gl:10.1.0, “react-native-unimodules”: “^0.12.0”,
Platforms(Android/iOS/web/all):Android
I am trying to add expo-three and expo-gl to my existing react-native app using unimodules

I followed the instructions here: Installing react-native-unimodules - Expo Documentation 1 and was able to get it to work on ios (with a few more steps with the bridge delegate) but I cannot get it working on android.

I end up with this error

app/node_modules/expo-gl/android/src/main/java/expo/modules/gl/GLObject.java:3: error: package expo.modules.gl.cpp does not exist
If i remove expo-gl then this will all compile, so it is something to do with that package.

I have these versions:
“expo-gl”: “^10.1.0”,
“expo-gl-cpp”: “^10.0.0”,
“expo-three”: “^5.6.2”,
“react-native-unimodules”: “^0.12.0”,
```

buildscript {
    repositories {
        maven { url 'https://plugins.gradle.org/m2/'}
    }
    dependencies {
        classpath 'gradle.plugin.com.onesignal:onesignal-gradle-plugin:[0.12.4, 0.99.99]'
    }
}
apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'
apply from: '../../node_modules/react-native-unimodules/gradle.groovy'
repositories {
    maven { url 'https://maven.google.com' }
}

buildscript {
    repositories {

        mavenCentral()
    }


    
}
apply plugin: "com.android.application"

repositories {
    google()
    mavenCentral()

}

import com.android.build.OutputFile


repositories {
    mavenCentral()
}

project.ext.react = [
    enableHermes: false
]

apply from: "../../node_modules/react-native/react.gradle"

def enableSeparateBuildPerCPUArchitecture = false

def enableProguardInReleaseBuilds = false

def useIntlJsc = false

android {
    signingConfigs {
        release {
            storeFile file(MYAPP_RELEASE_STORE_FILE)
            storePassword MYAPP_RELEASE_STORE_PASSWORD
            keyAlias MYAPP_RELEASE_KEY_ALIAS
            keyPassword MYAPP_RELEASE_KEY_PASSWORD
        }
    }
    defaultConfig {
        resConfigs "en", "US" , "es"

    }
    compileSdkVersion rootProject.ext.compileSdkVersion

    dexOptions {
        javaMaxHeapSize "3g"
    } 
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        applicationId "com.myapp"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 109
        versionName "1.8.12"
        multiDexEnabled true
         manifestPlaceholders = [onesignal_app_id: "<ID>",
                                onesignal_google_project_number: "REMOTE"]
        ndk {
            abiFilters  "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
    }
    buildTypes {
        release {
            // Caution! In production, you need to generate your own keystore file.
            // see https://facebook.github.io/react-native/docs/signed-apk-android.
            minifyEnabled enableProguardInReleaseBuilds
            signingConfig signingConfigs.release
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
       
    }

    packagingOptions {
        pickFirst "lib/armeabi-v7a/libc++_shared.so"
        pickFirst "lib/arm64-v8a/libc++_shared.so"
        pickFirst "lib/x86/libc++_shared.so"
        pickFirst "lib/x86_64/libc++_shared.so"
    }

    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    implementation fileTree(dir: "libs", include: ["*.jar"])
    //noinspection GradleDynamicVersion
    implementation "com.facebook.react:react-native:+"  // From node_modules
    implementation 'com.android.support:multidex:1.0.3'
    implementation 'androidx.annotation:annotation:1.0.2'
    implementation 'com.facebook.soloader:soloader:0.5.1'
   
    implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"


    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
      exclude group:'com.facebook.fbjni'
    }
    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }
    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }
    
    implementation 'com.onesignal:OneSignal:[3.11.2, 3.99.99]'
    
    implementation 'com.google.firebase:firebase-analytics:17.2.2'
    implementation 'com.crashlytics.sdk.android:crashlytics:2.10.1'
    implementation 'com.facebook.android:facebook-android-sdk:[5,6)'
    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'


     addUnimodulesDependencies([
            exclude      : [
                    // You can exclude unneeded modules here, e.g.,
                    // 'unimodules-face-detector-interface',
                    'expo-av',
                    
                    'expo-image-loader',
                    'expo-permissions',
                    'unimodules-image-loader-interface',
                    'unimodules-permissions-interface',
                    'unimodules-sensors-interface',
                    'unimodules-task-manager-interface',
                    'unimodules-face-detector-interface',
                    'unimodules-barcode-scanner-interface',
                    'unimodules-font-interface',
                    'unimodules-camera-interface',
                    // Adding a name here will also remove the package
                    // from auto-generated BasePackageList.java
            ]
    ])

    // JSC from node_modules
    if (useIntlJsc) {
        implementation 'org.webkit:android-jsc-intl:+'
    } else {
        implementation 'org.webkit:android-jsc:+'
    }

}
 
// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'
apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
apply plugin: 'com.google.gms.google-services'
apply plugin: 'io.fabric'
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

_No response_

### Environment

n/a

### Reproducible demo or steps to reproduce from a blank project

See gradle file above, followed all steps to install unimodules in an existing react-native project and successfully got ios to work but not android.

I tried posting to the forums https://forums.expo.io/t/cannot-build-expo-gl-using-unimodules/50662 and go no response so I'm not sure how best to figure this out.

I was able to recreate with this minimal repository https://github.com/jedahan/expo-sdk42-expo-gl use eas build https://expo.io/accounts/jedahan-reaktor/projects/sdk42-expo-gl/builds/ab335f87-c2c2-4f03-abf7-1c39c7b262fe with the following error:
```
Running './gradlew :app:bundleRelease' in /build/workingdir/build/android

Downloading https://services.gradle.org/distributions/gradle-6.8-all.zip

Unzipping /root/.gradle/wrapper/dists/gradle-6.8-all/9emh0d32e025cam7aasxcw002/gradle-6.8-all.zip to /root/.gradle/wrapper/dists/gradle-6.8-all/9emh0d32e025cam7aasxcw002

Set executable permissions for: /root/.gradle/wrapper/dists/gradle-6.8-all/9emh0d32e025cam7aasxcw002/gradle-6.8/bin/gradle

Welcome to Gradle 6.8!

Here are the highlights of this release:

 - Faster Kotlin DSL script compilation

- Vendor selection for Java toolchains

 - Convenient execution of tasks in composite builds

 - Consistent dependency resolution

For more details see https://docs.gradle.org/6.8/release-notes.html

To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/6.8/userguide/gradle_daemon.html#sec:disabling_the_daemon.

Daemon will be stopped at the end of the build

> Configure project :app

Installing unimodules:

 unimodules-core@7.1.1 from /build/workingdir/build/node_modules/@unimodules/core

 unimodules-react-native-adapter@6.3.4 from /build/workingdir/build/node_modules/@unimodules/react-native-adapter

 expo-application@3.2.0 from /build/workingdir/build/node_modules/expo-application

 expo-constants@11.0.1 from /build/workingdir/build/node_modules/expo-constants

 expo-error-recovery@2.2.0 from /build/workingdir/build/node_modules/expo-error-recovery

 expo-file-system@11.1.3 from /build/workingdir/build/node_modules/expo-file-system

 expo-font@9.2.1 from /build/workingdir/build/node_modules/expo-font

 expo-gl-cpp@10.4.0 from /build/workingdir/build/node_modules/expo-gl-cpp

 expo-gl@10.4.1 from /build/workingdir/build/node_modules/expo-gl

 expo-image-loader@2.2.0 from /build/workingdir/build/node_modules/expo-image-loader

 expo-keep-awake@9.2.0 from /build/workingdir/build/node_modules/expo-keep-awake

 expo-modules-core@0.2.0 from /build/workingdir/build/node_modules/expo-modules-core

 expo-splash-screen@0.11.2 from /build/workingdir/build/node_modules/expo-splash-screen

 expo-structured-headers@1.1.1 from /build/workingdir/build/node_modules/expo-structured-headers

expo-updates-interface@0.2.2 from /build/workingdir/build/node_modules/expo-updates-interface

 expo-updates@0.8.1 from /build/workingdir/build/node_modules/expo-updates

 unimodules-app-loader@2.2.0 from /build/workingdir/build/node_modules/unimodules-app-loader

 unimodules-task-manager-interface@6.2.0 from /build/workingdir/build/node_modules/unimodules-task-manager-interface

[stderr] FAILURE: Build completed with 2 failures.

[stderr] 1: Task failed with an exception.

[stderr] -----------

[stderr] * Where:

[stderr] Build file '/build/workingdir/build/node_modules/expo-gl-cpp/android/build.gradle' line: 2

[stderr] * What went wrong:

[stderr] A problem occurred evaluating project ':expo-gl-cpp'.

[stderr] > Plugin with id 'kotlin-android' not found.

[stderr] * Try:

[stderr] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

[stderr] ==============================================================================

[stderr] 2: Task failed with an exception.

[stderr] -----------

[stderr] * What went wrong:

[stderr] A problem occurred configuring project ':expo-gl-cpp'.

[stderr] > compileSdkVersion is not specified. Please add it to build.gradle

[stderr] * Try:

[stderr] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

[stderr] ==============================================================================

[stderr] * Get more help at https://help.gradle.org

[stderr] BUILD FAILED in 41s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.

Use '--warning-mode all' to show the individual deprecation warnings.

See https://docs.gradle.org/6.8/userguide/command_line_interface.html#sec:command_line_warnings

Error: Gradle build failed with unknown error. Please see logs for the "Run gradlew" phase.

Build failed: Gradle build failed with unknown error. Please see logs for the "Run gradlew" phase.
```

--- a/packages/expo-gl-cpp/android/build.gradle
+++ b/packages/expo-gl-cpp/android/build.gradle
@@ -17,6 +17,7 @@ buildscript {
 
   dependencies {
     classpath("de.undercouch:gradle-download-task:${safeExtGet("gradleDownloadTaskVersion", "3.4.3")}")
+    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${safeExtGet('kotlinVersion', '1.4.21')}")
   }
 }
 

